croud/session.py

```
@@ -65,22 +65,21 @@ def __init__(
     ) -> JsonDict:
         url = self.url + endpoint
         resp = await self.client.post(
-            url, json={"query": query, "variables": variables}
+            url, json={"query": query, "variables": variables}, allow_redirects=False
         )
-        if resp.status == 200:
-            try:
-                result = await resp.json()
-                return result
-            except ContentTypeError:
-                print_error(
-                    "Unauthorized. Use `croud login` to login to CrateDB Cloud."
-                )
-                await self.client.close()
-                exit(1)
-        else:
-            raise Exception(
-                f"Query failed to run by returning code of {resp.status}. {query}"
-            )
+
+        if resp.status == 302:  # login redirect
+            print_error("Unauthorized. Use `croud login` to login to CrateDB Cloud.")
+            await self.client.close()
+            exit(1)
```

This feels weird. Can we instead raise an exception that we catch on a higher level? Also, please use [`sys.exit()`](https://docs.python.org/3/library/sys.html#sys.exit).